File: backend/app/schemas/project.py
from datetime import datetime
from pydantic import BaseModel, Field
from typing import Optional

class ProjectCreate(BaseModel):
    title: str
    description: str | None = None
    okr_objectives: str | None = None
    key_results: str | None = None
    key_actions: str | None = None
    resources: str | None = None
    five_whys: str | None = None
    measurement_methods: str | None = None
    collaborators: list[str] = Field(default_factory=list)

class ProjectRead(BaseModel):
    id: int
    title: str
    description: str | None
    okr_objectives: str | None
    key_results: str | None
    key_actions: str | None
    resources: str | None
    five_whys: str | None
    measurement_methods: str | None
    submitted_by_name: str | None = None
    collaborators: list[str] = Field(default_factory=list)
    status: str
    source: str
    owner_id: int
    ms_list_id: str | None
    approved_by: int | None = None
    approved_at: datetime | None = None
    final_approved_by: int | None = None
    final_approved_at: datetime | None = None
    created_at: datetime
    updated_at: datetime
    model_config = {"from_attributes": True}

class AprobacionFinalInput(BaseModel):
    salario_base: float
    cargo: str
    num_personas: int = 1
    sede: str | None = None
    observacion: str | None = None

class CustomQuestionInput(BaseModel):
    text: str
    axis: str = "impact"  # "impact" | "effort"

class SuperaprobacionInput(BaseModel):
    question_ids: list[int] = []
    custom_questions: list[CustomQuestionInput] = []  # ← objetos con axis, no strings

class SalarioInput(BaseModel):
    salario_base: float
    cargo: str
    sede: str | None = None

class DatosOperacionalesInput(BaseModel):
    num_personas: int
    horas_proceso_actual: float
    horas_proceso_nuevo: float
    observacion: str | None = None

class ProjectQuestionRead(BaseModel):
    id: int
    project_id: int
    question_text: str
    axis: str = "impact"           # ← AÑADIDO: "impact" | "effort"
    source_question_id: int | None = None   # ← eliminado el duplicado
    created_by: int
    created_at: datetime
    model_config = {"from_attributes": True}

File: frontend/src/types/project.ts
// frontend/src/types/project.ts
export interface Project {
  id: number
  title: string
  description: string | null
  okr_objectives: string | null
  key_results: string | null
  key_actions: string | null
  resources: string | null
  five_whys: string | null
  measurement_methods: string | null
  submitted_by_name: string | null
  okr_creator: string | null
  collaborators: string[]
  due_date: string | null
  okr_productive: boolean | null
  status: string
  source: string
  owner_id: number
  /** Área funcional del usuario dueño (catálogo); útil para filtrar vistas operativas */
  owner_work_area_id?: number | null
  owner_work_area_name?: string | null
  ms_list_id: string | null
  approved_by: number | null
  approved_at: string | null
  final_approved_by: number | null
  final_approved_at: string | null
  assigned_to_dev: boolean
  assigned_to_dev_at: string | null
  assigned_to_dev_by: number | null
  assigned_area_id: number | null
  assigned_area_at: string | null
  assigned_area_by: number | null
  assigned_area_name: string | null
  created_at: string
  updated_at: string
  /** Conteo de evidencias activas; ausente se trata como 0 */
  evidence_count?: number
  /** Calculado en backend: owner o equipo dev si assigned_to_dev */
  viewer_can_modify_tasks?: boolean
}

export interface ProjectCreate {
  title: string
  description?: string
  okr_objectives?: string
  key_results?: string
  key_actions?: string
  resources?: string
  five_whys?: string
  measurement_methods?: string
  okr_creator?: string
  collaborators?: string[]
}

export interface ProjectQuestion {
  id: number
  project_id: number
  question_text: string
  axis: "impact" | "effort"   
  source_question_id: number | null
  created_by: number
  created_at: string
}

// Payload para que el superadmin apruebe y asigne preguntas
export interface CustomQuestionPayload {
  text: string
  axis: "impact" | "effort"
}

export interface SuperaprobacionPayload {
  question_ids: number[]
  custom_questions: CustomQuestionPayload[]
}

// Payload para que el superadmin provea el salario
export interface SalarioPayload {
  salario_base: number
  cargo: string
  sede?: string
}

// Payload para que el admin complete datos operacionales del ROI
export interface DatosOperacionalesPayload {
  num_personas: number
  horas_proceso_actual: number
  horas_proceso_nuevo: number
  observacion?: string
}

// Payload para que usuario/coordinador extienda la fecha de su propio proyecto
export interface DueDateExtendPayload {
  due_date: string // YYYY-MM-DD
  justificacion: string
}
